### PR TITLE
[syncer] change BestPeer to BetterPeer

### PR DESCRIPTION
**Pull request by 0xcb9ff9**
# Description

Randomly choose a node higher than itself, reduce the pressure on the highest node

# Changes include

- [x] New feature (non-breaking change that adds functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels

## Testing

- [x] I have tested this code with the official test suite

### CODE REVIEW

**Review by DarianShawn — DISMISSED** [review]
LGTM.

File: protocol/syncer.go [hunk 1]
```diff
@@ -305,23 +305,23 @@ func (s *noForkSyncer) syncWithSkipList(
 	}
 
 	// if the bestPeer does not have a new block continue
-	if bestPeer.Number <= localLatest {
-		s.logger.Debug("wait for the best peer catching up the latest block", "bestPeer", bestPeer.ID)
+	if betterPeer.Number <= localLatest {
+		s.logger.Debug("wait for the best peer catching up the latest block", "bestPeer", betterPeer.ID)
```
Comment (DarianShawn) [inline protocol/syncer.go]: `bestPeer` => `betterPeer`

File: protocol/peer.go [hunk 2]
```diff
@@ -50,9 +51,30 @@ func (m *PeerMap) Remove(peerID peer.ID) {
 	m.Delete(peerID.String())
 }
 
-// BestPeer returns the top of heap
-func (m *PeerMap) BestPeer(skipMap *sync.Map) *NoForkPeer {
-	var bestPeer *NoForkPeer
+// ToList return all peers (sort by IsBetter)
+func (m *PeerMap) ToList() []*NoForkPeer {
```
Comment (DarianShawn) [inline protocol/peer.go]: The test helper method should be in `peer_test.go`

**Review by DarianShawn — COMMENTED** [review]
Only a few suggestions.

**Comment by 0xcb9ff9** [comment]
PR #305 replace it

**Pull request closed without merging**